packages/react-dom/src/client/ReactDOMComponentTree.js

```diff
@@ -40,17 +37,13 @@ export function getClosestInstanceFromNode(node) {
     }
   }
 
-  let closest;
   let inst = node[internalInstanceKey];
   if (inst.tag === HostComponent || inst.tag === HostText) {
     // In Fiber, this will always be the deepest root.
     return inst;
   }
-  for (; node && (inst = node[internalInstanceKey]); node = parents.pop()) {
-    closest = inst;
-  }
```

Comment:
I've tried to check back through the history to see what was going on here, and at least originally it used to try and precache nodes: https://github.com/facebook/react/commit/6d20556c78df6904254b8438bb588f7722ea602f#diff-a2fd175216a5aa6425363e6773426f3c

But at some point that was removed but I can't find git blame is not helping :/

Then

Reply:
When does line 50 trigger?

Reply:
Sorry, I mean like what code path causes closest to be assigned to the instance? And what codepath even lets this function get past `return inst` on line 47?

Reply:
I think in practical usage it doesn't get past 47, but I guess theoretically it could, if the fiber was a different type?

My guess tho is that this was for both stack and fiber and never got removed. Though even if it could get past 47 I don't think the loop does anything other than run one iteration, and return `inst`

Reply:
Cool. I'd love to drop this if the code path no longer fires after Stack's removal. I'm afraid to ping anyone to bug them on holiday, but I think it's worth investigating.

Reply:
no rush clearly. :)

Reply:
It was hard to find due to a couple project reorganizations nuking the file history, but this was removed in https://github.com/facebook/react/pull/10798. I think the most likely explanation is that the whole loop could have been removed as well, but it wasn't clear that it was stack-only.

It's a little confusing, but at first glance it seems safe to remove 😄